### PR TITLE
added logical flags for state variable outputs

### PR DESCRIPTION
**Pull request by epiyasin**
Clear large number of un-used state variables to thin model output. Introduces flag logic, notably in the form of being able to switch on/off the following; rt, cum_admit, diagnoses_admitted, cum_infections_disag, cum_n_vaccinated. Currently with all the state variables turned on (i.e = TRUE) the output state trajectories are 816. Further PRs for spimalot https://github.com/mrc-ide/spimalot/pull/181 and ncov https://github.com/ncov-ic/ncov-outputs/pull/2064 focus on implementing these changes within the fitting and simulation pipeline. Functionality for switches to be turned off will come in later PRs.

### CODE REVIEW

File: R/lancelot.R [hunk 1]
```diff
@@ -711,6 +711,23 @@ process_strain_rel_p <- function(p, n_strains, n_real_strains) {
 ##'
 ##' @inheritParams basic_index
 ##'
+##' @param rt logical parameter output trajectories required for
```
Comment (edknock) [inline R/lancelot.R:714]:
```suggestion
##' @param rt Logical, whether to output trajectories required for
```

File: R/lancelot.R [hunk 2]
```diff
@@ -711,6 +711,23 @@ process_strain_rel_p <- function(p, n_strains, n_real_strains) {
...
+##'   calculating Rt (default = FALSE)
+##'
+##' @param cum_admit logical parameter whether to output cumulative
```
Comment (edknock) [inline R/lancelot.R:717]:
```suggestion
##' @param cum_admit Logical, whether to output cumulative
```

File: R/lancelot.R [hunk 3]
```diff
@@ -711,6 +711,23 @@ process_strain_rel_p <- function(p, n_strains, n_real_strains) {
...
+##'   admissions by age (default = FALSE)
+##'
+##' @param diagnoses_admitted logical parameter whether to output
```
Comment (edknock) [inline R/lancelot.R:720]: ****
```suggestion
##' @param diagnoses_admitted Logical, whether to output
```

File: R/lancelot.R [hunk 4]
```diff
@@ -711,6 +711,23 @@ process_strain_rel_p <- function(p, n_strains, n_real_strains) {
...
+##'   cumulative combined confirmed admissions and inpatient
+##'   diagnoses by age and vaccine class (default = FALSE)
+##'
+##' @param cum_infections_disag logical parameter whether to output
```
Comment (edknock) [inline R/lancelot.R:724]:
```suggestion
##' @param cum_infections_disag Logical, whether to output
```

File: R/lancelot.R [hunk 5]
```diff
@@ -711,6 +711,23 @@ process_strain_rel_p <- function(p, n_strains, n_real_strains) {
...
+##'   cumulative infections by age and vaccine class (default = FALSE)
+##'
+##' @param cum_n_vaccinated logical parameter whether to output
```
Comment (edknock) [inline R/lancelot.R:727]:
```suggestion
##' @param cum_n_vaccinated Logical, whether to output
```

**Comment by epiyasin** [comment]
codecov failing due to pre-existing miss on master

File: tests/testthat/test-lancelot-pmcmc.R [hunk 6]
```diff
@@ -37,10 +30,80 @@ test_that("can add and remove trajectories from mcstate_pmcmc objects", {
...
-  expect_identical(res$state["deaths_inc", , ],
-                   cmp$state["deaths_inc", , ])
+  cmp <- add_trajectory_incidence(dat$trajectories, c("general", "icu"))
+  res <- add_trajectory_incidence(dat$trajectories, "general")
+  expect_identical(res$state["general_inc", , ],
+                   cmp$state["general_inc", , ])
+})
+
+
+test_that("Can drop predictions from trajectories", {
```
Comment (edknock) [inline tests/testthat/test-lancelot-pmcmc.R:40]: Remove this test and the next 4 as they've been deleted from master (they are testing functions that have been deleted)

File: tests/testthat/helper-lancelot-pmcmc.R [hunk 7]
```diff
@@ -54,14 +54,14 @@ reference_data_lancelot_mcmc <- function() {
 reference_data_lancelot_trajectories <- function() {
   load_reference("data/lancelot_trajectories.rds", {
     dat <- reference_data_lancelot_mcmc()
-    incidence <- "deaths"
+    incidence <- "icu"
```
Comment (richfitz) [inline tests/testthat/helper-lancelot-pmcmc.R:57]: I believe these changes can be reverted - can you please check

File: tests/testthat/helper-lancelot.R [hunk 8]
```diff
@@ -35,13 +35,18 @@ helper_lancelot_data <- function(data, start_date, dt) {
 helper_lancelot_particle_filter <- function(data, n_particles,
                                             n_threads = 1L, seed = NULL,
                                             compiled_compare = FALSE) {
+
+  fit_index <- function(info) {
```
Comment (richfitz) [inline tests/testthat/helper-lancelot.R:39]: I believe these changes can be reverted

File: tests/testthat/test-lancelot-pmcmc.R [hunk 9]
```diff
@@ -26,7 +26,7 @@ test_that("adding incidence adds appropriate states", {
 
 test_that("can add and remove trajectories from mcstate_pmcmc objects", {
   dat <- reference_data_lancelot_mcmc()
-  v <- c("deaths", "icu")
+  v <- c("general", "icu")
```
Comment (richfitz) [inline tests/testthat/test-lancelot-pmcmc.R:29]: I believe these can be reverted

**Review by richfitz — CHANGES_REQUESTED** [review]
I'd like you to back out some of the changes that were made to the tests but appear no longer needed.  You can do this by running

```
git fetch
git merge origin/fit_index-function-patch
```

or by working through the comments I've left below

**Review by richfitz — APPROVED** [review]

**Comment by richfitz** [comment]
buildkite build taking its time due to slow archive downloading, will pass eventually

**Pull request merged**